dipy/core/gradients.py

```diff
@@ -134,11 +134,11 @@ def gradient_table_from_bvals_bvecs(bvals, bvecs, b0_threshold=0, atol=1e-2,
                          "respectively, where N is the number of diffusion "
                          "gradients")
 
+    bvecs[np.isnan(bvecs)] = 0
```

Comment:
Please use something like `bvecs = np.where(np.isnan(bvecs), 0, bvecs)`, `bvecs` is an argument to the function, we don't want to silently modify the function arguments.